pokemon_search/src/index.js

```diff
@@ -1,14 +1,23 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import reportWebVitals from "./reportWebVitals";
+import Details from "./routes/details";
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<App />} />
```

Comment:
We are using `App` as the homepage here. This is fine but could raise some questions for the reader about the app structure.

I think an alternative way to structure this is to move the existing logic in `App` to a new component called `Home`.
Then move this `react-router` logic to `App`.

So, here's the proposed responsibilities for each component:
- `App.js` contains the `react-router` logic. We can easily see the overview/route structure by going to App
- `Home.js` contains the home page logic
- `Details.js` contains the individual pokemon logic
- `index.js` just calls `ReactDOM` to mount the `App` ( similar to before )

How do you feel about this structure?

Reply:
I like it. More logical.